src/VisualStudio/Core/Def/Implementation/Preview/PreviewUpdater.PreviewDialogWorkspace.cs

```diff
@@ -25,25 +26,43 @@ public void UnregisterTextContainer(SourceTextContainer container)
 
             public void CloseDocument(TextDocument document, SourceText text)
             {
-                if (document is Document)
+                switch (document.Kind)
```

Comment:
if we have this. it feels like we can just have 1 helper method to add, remove, change text on these document rather than having lambda for those?

Reply:
Open and close as well.

Reply:
Yep, good suggestions. However, I have tried to minimize the changes in this PR by avoiding any big shimming/code refactorings as much as possible.